internal/cmd/stop.go

```diff
@@ -17,15 +21,53 @@ func SetupStopCli() *cobra.Command {
 		Short: "Stop CNTLM Proxy",
 		Run:   stopCmd,
 	}
+	stopCmd.Flags().Bool("all", true, "unset all proxy configuration")
 	return stopCmd
 }
 
 func stopCmd(cmd *cobra.Command, args []string) {
-	profile.UpdateGlobalEnvironmentVariables("")
+	cfg, err := config.LoadConfiguration(proxyProfile)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if cmd.Flags().Changed("all") {
+		if err = unsetAllConfiguration(&cfg); err != nil {
+			log.Fatal(err)
+		}
+	}
+	emptyURL, err := url.Parse("")
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = env.UpdateGlobalEnvironmentVariables(emptyURL)
+	if err != nil {
+		log.Fatal(err)
+	}
 	cmds := execute.Command{Cmd: "pkill", Args: []string{"cntlm"}}
-	_, err := execute.RunCommand(cmds)
+	_, err = execute.RunCommand(cmds)
 	if err != nil {
 		log.Fatalf("Couldn't kill CNTLM process %q", err)
 	}
+	viper.Set("Proxy.Running", false)
+	err = config.SaveConfiguration(proxyProfile)
+	if err != nil {
+		log.Fatal(err)
+	}
 	fmt.Println("CNTLM proxy stopped")
 }
+
+func unsetAllConfiguration(cfg *config.Configuration) error {
+	// TODO: Consider a better way to loop through Tools as the list of supported CLI's may grow
```

Comment:
One way that comes to mind would be to create an interface for plugins. The plugin interface could have functions like `UnsetAll`, `CreateCobraCmd` and `SetAll`.